artemis-journal/src/main/java/org/apache/activemq/artemis/core/io/buffer/TimedBuffer.java

```diff
@@ -93,10 +82,10 @@
    public TimedBuffer(final int size, final int timeout, final boolean logRates) {
       bufferSize = size;
 
-      this.logRates = logRates;
-
       if (logRates) {
-         logRatesTimer = new Timer(true);
+         this.flushProfiler = Profiler.instrumented();
+      } else {
+         this.flushProfiler = Profiler.none();
```

Comment:
About the compiler we're "lucky", because the noop impl will be inlined (is monomorphic with no code inside) and then wiped away (verified using JitWatch), leaving the TimedBuffer clean and without null checks too.
The guard statement that can be compiled out AFAIK are the ones that rely on final static fields, while normal instance members do not receive the same treatment from the JIT, sadly.
It is a different coding style: like having a null logger instance member or a noop logger implementation...wdyt?